nominal/core.py

```diff
@@ -624,7 +612,7 @@ def create_dataset_from_io(
         dataset: BinaryIO,
         name: str,
         timestamp_column: str,
-        timestamp_type: TimestampColumnType,
+        timestamp_type: TypedTimeDomain,
```

Comment:
the docs are now out of date. Also, need to describe the functionality here in multiple places.